### PR TITLE
Optimize api for fetching user's selected library

### PR DESCRIPTION
**Pull request by mileswong**

### CODE REVIEW

File: modmod/views/library.py [hunk 1]
```diff
@@ -81,13 +81,13 @@ def list_library(request):
...
-    query_types = request.GET.get('type')
+    query_types = request.GET.getall('type')
     if query_types:
-        query_types = query_types.lower()
+        query_types = [type.lower() for type in query_types]
 
     # if type is not specified, return all types of library data
     all_types = ['public', 'private', 'forsale', 'selected', 'unselected']
-    query_types = query_types.split(" ") if query_types else all_types
+    query_types = query_types or all_types
```
Comment (nwingt) [inline modmod/views/library.py]: Simplify into this
```python
...
else:
    query_types = ['public', 'private', 'forsale', 'selected', 'unselected']
...
```

File: modmod/models/library.py [hunk 2]
```diff
@@ -123,7 +123,6 @@ def serialize(self, user=None):
         serialized_library['author'] = self.users[0].serialize_min() if self.users else None  # Assume user[0] is the author
         serialized_library['updatedAt'] = self.updated_at.isoformat()
         serialized_library['launchedAt'] = self.launched_at.isoformat() if self.launched_at else None
-        serialized_library['isSelected'] = self.has_user_selected(user)
```
Comment (nwingt) [inline modmod/models/library.py]: leave `serialized_library['isSelected']` here and set default in the function arguments

i.e.
```python
def serialize(self, ... is_selected=False)

serialized_library['isSelected'] = is_selected
```

File: modmod/views/library.py [hunk 3]
```diff
@@ -123,17 +121,11 @@ def list_library(request):
         elif t == 'selected':
```
Comment (nwingt) [inline modmod/views/library.py]: Suppose public, private and forSale are selected?

Comment (mileswong) [reply modmod/views/library.py]: `selected` here means selected libraries that are created by other users;
want to change `selected` and `unselected` for more clarified key names?

**Pull request merged**